base_custom_attributes/custom_attributes.py

```diff
@@ -0,0 +1,449 @@
+                return super(AttributeAttribute, self).create(vals)
+
+        if vals.get('relation_model_id'):
+            relation = self.env['ir.model'].browse(
+                vals.get('relation_model_id')).model
+        else:
+            relation = 'attribute.option'
+        if vals['attribute_type'] == 'select':
+            vals['ttype'] = 'many2one'
+            vals['relation'] = relation
+        elif vals['attribute_type'] == 'multiselect':
+            vals['ttype'] = 'many2many'
+            vals['relation'] = relation
+            vals['serialized'] = True
+        else:
+            vals['ttype'] = vals['attribute_type']
+
+        if vals.get('serialized'):
+            field_obj = self.pool.get('ir.model.fields')
+            serialized_ids = field_obj.search([
+                ('ttype', '=', 'serialized'),
+                ('model_id', '=', vals['model_id']),
+                ('name', '=', 'x_custom_json_attrs')])
+            if serialized_ids:
+                vals['serialization_field_id'] = serialized_ids[0]
+            else:
+                f_vals = {
+                    'name': u'x_custom_json_attrs',
+                    # 'field_description': u'Serialized JSON Attributes',
+                    'ttype': 'serialized',
+                    'model_id': vals['model_id'],
+                }
+                vals['serialization_field_id'] = field_obj.create(
+                    f_vals, {'manual': True})
+        vals['state'] = 'manual'
+        return super(AttributeAttribute, self).create(vals)
+
+    @api.onchange('field_description')
+    def onchange_field_description(self):
+        name = self.name or u'x_'
+        if self.field_description and not self.create_date:
+            name = unicode('x_' + safe_column_name(self.field_description))
+        self.name = name
+
+    @api.onchange('name')
+    def onchange_name(self):
+        res = {}
+        if not self.name.startswith('x_'):
+            name = u'x_%s' % self.name
+        else:
+            name = u'%s' % self.name
+        res = {'value': {'name': unidecode(name)}}
+
+        context = self._context
+        model_name = context.get('force_model')
+        if not model_name:
+            model_id = context.get('default_model_id')
+            if model_id:
+                model = self.env['ir.model'].browse(model_id)
+                model_name = model.model
+        if model_name:
+            model_obj = self.env[model_name]
+            allowed_model = [x for x in model_obj._inherits] + [model_name]
+            res['domain'] = {'model_id': [['model', 'in', allowed_model]]}
+
+        return res
+
+
+class AttributeGroup(models.Model):
+    _name = "attribute.group"
+    _description = "Attribute Group"
+    _order = "sequence"
+
+    @api.model
+    def _get_default_model(self):
+        context = self._context
+        if context and context.get('force_model'):
+            model_id = self.env['ir.model'].search(
```

Comment:
Please rename this variable as you are not getting an id but a record set, plus, add `limit=1` as you are looking for a single result